lib/MPIPreferences/src/parse_cray_cc.jl

```diff
@@ -0,0 +1,94 @@
+module CrayParser
+
+import Base: filter, map, reduce
+filter(f::Function)::Function = x -> filter(f, x)
+map(f::Function)::Function    = x -> map(f, x)
+reduce(f::Function)::Function = x -> reduce(f, x)
```

Comment:
I think it's better to use `Base.Fix1(filter)` and such where you need to use these functions instead of adding methods to Base functions.

Reply:
makes sense -- done